lib/defender/aws_iot_defender_collector.c

```diff
@@ -358,6 +359,7 @@ static void tcpConnectionsCallback( void * param1,
     IotMetricsTcpConnection_t * pConnection = NULL;
 
     size_t total = IotListDouble_Count( pTcpConnectionsMetricsList );
+    uint8_t i = 0;
```

Comment:
Same... use size_t or  uint32_t to count...

Reply:
Or use `BaseType_t` if `FreeRTOS.h` is included. `BaseType_t` promises that the width is equal to the size of a register.